### ISSUE TITLE
ORTE has lost communication with a remote daemon.

### ISSUE THREAD

**Issue by tingweiwu** [open-mpi/ompi#6618]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v1.10.3, v2.1.0, git branch name and hash, etc.)

```
 mpirun --version
mpirun.real (OpenRTE) 3.1.2
```

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

Install Open MPI
```
 mkdir /tmp/openmpi && \
    cd /tmp/openmpi && \
    wget https://www.open-mpi.org/software/ompi/v3.1/downloads/openmpi-3.1.2.tar.gz && \
    tar zxf openmpi-3.1.2.tar.gz && \
    cd openmpi-3.1.2 && \
    ./configure --enable-orterun-prefix-by-default && \
    make -j $(nproc) all && \
    make install && \
    ldconfig && \
    rm -rf /tmp/openmpi
```

### Please describe the system on which you are running

* Operating system/version: 
ubuntu16.04
* Computer hardware: 
V100GPU+InfiniBand
* Network type: 
docker cni networker

-----------------------------

## Details of the problem

Please describe, in detail, the problem that you are having, including the behavior you expect to see, the actual behavior that you are seeing, steps to reproduce the problem, etc.  It is most helpful if you can attach a small program that a developer can use to reproduce your problem.

**Note**: If you include verbatim output (or a code block), please use a [GitHub Markdown](https://help.github.com/articles/creating-and-highlighting-code-blocks/) code block like below:

I got this error frequently, not everytime. but it occures  both when the process starting or running.

 I have check the network bewteen ` i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd` and `i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0`  is ok, and OOM haven't occured.

do you have any suggestion to find the reason?

```shell
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 1 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 3 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 2 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 4 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 5 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 6 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 7 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
+ POD_NAME=i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7
+ shift
+ /opt/kube/kubectl exec i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7 -- /bin/sh -c PATH=/usr/local/bin:$PATH ; export PATH ; LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH ; export LD_LIBRARY_PATH ; DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH ; export DYLD_LIBRARY_PATH ; /usr/local/bin/orted -mca ess "env" -mca ess_base_jobid "745603072" -mca ess_base_vpid 8 -mca ess_base_num_procs "9" -mca orte_node_regex "i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-1,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-2,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-3,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-4,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-5,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-6,i[5:39030]a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-7@0(9)" -mca orte_hnp_uri "745603072.0;tcp://192.168.237.70:59121" -mca plm "rsh" -mca plm_rsh_agent "/etc/mpi/kubexec.sh" -mca orte_default_hostfile "/etc/mpi/hostfile" -mca pmix "^s1,s2,cray,isolated"
command terminated with exit code 137
--------------------------------------------------------------------------
ORTE has lost communication with a remote daemon.

HNP daemon : [[11377,0],0] on node i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd
Remote daemon: [[11377,0],1] on node i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0

This is usually due to either a failure of the TCP network
connection to the node, or possibly an internal failure of
the daemon itself. We cannot recover from this failure, and
therefore will terminate the job.
--------------------------------------------------------------------------
ORTE has lost communication with a remote daemon.

HNP daemon : [[11377,0],0] on node i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-launcher-ksczd
Remote daemon: [[11377,0],1] on node i39030a4a38d4a3abcb17e488a3141ef-mpijob-0-worker-0

This is usually due to either a failure of the TCP network
connection to the node, or possibly an internal failure of
the daemon itself. We cannot recover from this failure, and
therefore will terminate the job.
--------------------------------------------------------------------------
```

**Comment by tingweiwu** [comment]
how could I get more infomation about this error?

**Comment by zrss** [comment]
i met a similar case, but i found my problem is related to the number of `np`, as i run

```
mpirun -np 192 hostname
```

it works fine, but once i increased the np number to 200, it comes out the problem

```
ORTE has lost communication with a remote daemon.

  HNP daemon   : [[2448,0],0] on node job1b-train-pub-v100nv2-ueaxr
  Remote daemon: [[2448,0],6] on node jobbf317b09-worker-13-6pg-862-gdwfm

This is usually due to either a failure of the TCP network
connection to the node, or possibly an internal failure of
the daemon itself. We cannot recover from this failure, and
therefore will terminate the job.
```

and i can confirm these nodes can connect each other as i switch the hostfile to select nodes that mpirun run

the mpi hostfile sample

```
jobbf317b09-worker-31-q8d-188-d5d9p slots=8
jobbf317b09-worker-4-ob8-102-9srjx slots=8
jobbf317b09-worker-5-c9p-866-nc6pb slots=8
jobbf317b09-worker-6-aqj-472-mdglq slots=8
jobbf317b09-worker-7-95f-872-667dz slots=8
jobbf317b09-worker-8-vci-856-7lsgq slots=8
jobbf317b09-worker-9-0hj-7-m9vgz slots=8
```

... hope that someone know how to debug it :alien:

**Comment by tingweiwu** [comment]
It's possible to configure the openmpi to not terminate the job if ORTE has lost communication with a remote daemon?

**Comment by tingweiwu** [comment]
```
io time cost (batch 2640):  252.29486799240112
epoch 4/100, batch: 2620/3461, lr: 0.00048600003, loss: 7.65098391976442, duration per batch: 25062.119s, memory: 6.0GB
--------------------------------------------------------------------------
ORTE has lost communication with a remote daemon.

  HNP daemon   : [[37833,0],0] on node orte-mpijob-0-launcher-6cskl
  Remote daemon: [[37833,0],2] on node orte-mpijob-0-worker-1

This is usually due to either a failure of the TCP network
connection to the node, or possibly an internal failure of
the daemon itself. We cannot recover from this failure, and
therefore will terminate the job.
--------------------------------------------------------------------------
[orte-mpijob-0-launcher-6cskl:00009] Job UNKNOWN has launched
[orte-mpijob-0-launcher-6cskl:00009] [[37833,0],0] Releasing job data for [37833,1]
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_finalize: proc session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_finalize: job session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_finalize: jobfam session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_finalize: jobfam session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_finalize: top session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_cleanup: job session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_cleanup: top session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] [[37833,0],0] Releasing job data for [37833,0]
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_cleanup: job session dir does not exist
[orte-mpijob-0-launcher-6cskl:00009] sess_dir_cleanup: top session dir does not exist
exiting with status -51

```

 I add `--debug-devel` to mpirun.  no more usefull infomation shows when it failed.

**Comment by jsquyres** [comment]
If Open MPI hasn't even finished launching yet and something goes wrong, that's a fatal error -- we don't have any other option except to abort.

Your job is relatively small -- 8 nodes.  You shouldn't be bumping up against any TCP sockets or file descriptor limits.

I'm assuming you're launching in Kubernetes -- is each one of these `/opt/kube/kubectl` launching into a unique container?  If so, are these containers fully distinct from each other -- e.g., different IP address, different PID space, ...?

**Comment by zrss** [comment]
@tingweiwu @jsquyres , hi, as to my problem, mpirun hostname failed with np 256, i found out a workaround, i.e. run mpirun with `--mca plm_rsh_no_tree_spawn true` or specific the routed module (radix, binomial, debruijn or direct)

#6691

**Comment by rhc54** [comment]
Either two or three is fine - no need for multi-select anyway

**Comment by jjhursey** [comment]
@zrss I belive that PR #6944 (maybe with PR #6915 for your environment) should fix this issue. If you have an opportunity can you try that patch to see if it resolves the issue that you were seeing with tree spawn?

**Comment by jjhursey** [comment]
Note: I checked the v3.0.x and v3.1.x branches and they do not show this specific problem. It must be unique to the v4.0.x branch.

**Comment by rhc54** [comment]
Yes, it is - the reason is that the v4.0 branch came during a point in time where we were trying to use libfabric for ORTE collectives during launch. This meant that the mgmt Ethernet OOB connections required a different routing pattern from the libfabric ones - and hence we made the routed framework multi-select. We decided not to pursue that path _after_ the v4.0 branch and offered to remove that code from the release branch, but it was deemed too big a change.

If it were me, I'd just make routed single-select and completely resolve the problem. Nothing will break because you cannot use the RML/OFI component unless you explicitly request it. However, there is nothing wrong with this approach.
